group-generators/helpers/data-providers/dune/provider.ts

```diff
@@ -0,0 +1,179 @@
+import { DuneErrorFactory } from './errors';
+import {
+  QueryParams,
+  ExecutionState,
+  ExecuteQuery,
+  ExecutionStatusComplete,
+  ExecutionResults,
+  Row,
+} from './types';
+import { FetchedData } from 'topics/group';
+
+//Dune requires an API key
+export class DuneProvider {
+  apikey: string;
+
+  public constructor() {
+    this.apikey = this.getApiKey();
+  }
+
+  /**
+   * Use this method to query Dune Analytics for your data.
+   * @param queryId The Dune QueryID e.g. 2034748 https://dune.com/queries/2034748"
+   * @param duneEthAddressColumn e.g. “The ETH Address Winner”  for the Dune Query https://dune.com/queries/2034748 -> used to match the ETH addresses
+   * @param queryParams optional! e.g. {address: '0x3CAaE25Ee616f2C8E13C74dA0813402eae3F496b', blocknumber: '66257668', chain: 'arbitrum'} for the Dune Query https://dune.com/queries/1618116
+   * @returns The FetchedData object in the <FetchedData> Type
+   */
+  public async executeQuery(
```

Comment:
Maybe you can put only `executeQuery()` and `executeQueryCount()` in public, and all the other methods in private